tests/test_views.py

```diff
@@ -28,406 +28,444 @@
+
+                res = client.get(
+                    url_for("workflows.get_workflows"),
+                    query_string={
+                        "access_token": default_user.access_token,
+                        "type": "batch"},
+                )
+                assert res.status_code == 200
 
 
 def test_create_workflow(app, default_user):
     """Test create_workflow view."""
     with app.test_client() as client:
-        with patch(
-            "reana_server.rest.workflows.current_rwc_api_client",
-            make_mock_api_client("reana-workflow-controller")(),
-        ):
-            # access token needs to be passed instead of user_id
-            res = client.post(
-                url_for("workflows.create_workflow"),
-                query_string={"user_id": default_user.id_},
-            )
-            assert res.status_code == 403
-
-            # remote repository given as spec, not implemented
-            res = client.post(
-                url_for("workflows.create_workflow"),
-                query_string={"access_token": default_user.access_token,
-                              "spec": "not_implemented"},
-            )
-            assert res.status_code == 501
-
-            # no specification provided
-            res = client.post(
-                url_for("workflows.create_workflow"),
-                query_string={"access_token": default_user.access_token},
-            )
-            assert res.status_code == 500
-
-            # unknown workflow engine
-            workflow_data = {
-                "workflow": {"specification": {}, "type": "unknown"},
-                "workflow_name": "test",
-            }
-            res = client.post(
-                url_for("workflows.create_workflow"),
-                headers={"Content-Type": "application/json"},
-                query_string={"access_token": default_user.access_token},
-                data=json.dumps(workflow_data),
-            )
-            assert res.status_code == 500
-
-            # name cannot be valid uuid4
-            workflow_data['workflow']['type'] = 'serial'
-            res = client.post(
-                url_for("workflows.create_workflow"),
-                headers={"Content-Type": "application/json"},
-                query_string={"access_token": default_user.access_token,
-                              "workflow_name": str(uuid4())},
-                data=json.dumps(workflow_data),
-            )
-            assert res.status_code == 400
+        mock_current_user = Mock(is_authenticated=False)
+        with patch('flask_login.utils._get_user',
+                   Mock(return_value=mock_current_user)):
+            with patch(
+                "reana_server.rest.workflows.current_rwc_api_client",
+                make_mock_api_client("reana-workflow-controller")(),
+            ):
+                # access token needs to be passed instead of user_id
+                res = client.post(
```

Comment:
We can remove this `access token needs to be passed instead of user_id ` part of the test.